src/Microsoft.OData.Core/Value/ODataStreamReferenceValue.cs

```diff
@@ -8,21 +8,108 @@ namespace Microsoft.OData
 {
     #region Namespaces
     using System;
+    using System.IO;
     using Microsoft.OData.Evaluation;
+    using Edm;
 
     #endregion
 
+    // todo (mikep): move these to their own files
+
     /// <summary>
-    /// Represents a media resource.
+    /// A class to represent a binary stream value
     /// </summary>
-    public sealed class ODataStreamReferenceValue : ODataValue
+    public sealed class ODataBinaryStreamValue : ODataValue
     {
-        /// <summary>the metadata builder for this OData resource.</summary>
-        private ODataResourceMetadataBuilder metadataBuilder;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="stream">Input stream</param>
+        public ODataBinaryStreamValue(Stream stream)
+        {
+            this.Stream = stream;
+        }
 
+        /// <summary>
+        /// The Stream wrapped by the ODataStreamValue
+        /// </summary>
+        public Stream Stream { get; private set; }
+    }
+
+    /// <summary>
+    /// A class to represent a stream value
+    /// </summary>
+    public class ODataStreamValue : ODataValue
+    {
         /// <summary>The name of the named stream this value belongs to; null for the default media resource.</summary>
         private string edmPropertyName;
 
+        /// <summary>Primitive Type, if known.  Either String or None.</summary>
+        private EdmPrimitiveTypeKind typeKind;
+
+        /// <summary>
+        /// Creates an ODataStringValue with a specified typeKind
+        /// </summary>
+        /// <param name="typeKind">The type of stream value</param>
+        internal ODataStreamValue(EdmPrimitiveTypeKind typeKind)
+        {
+            this.typeKind = typeKind;
+        }
+
+        /// <summary>
+        /// PrimitiveTypeKind of the stream.
+        /// </summary>
+        /// <returns>
+        /// EdmPrimitiveTypeKind.String, if the contents is a plain string;
+        /// EdmPrimitiveTypeKind.None if it could be a binary string

```

Comment:
>None [](start = 33, length = 4)

Should be "EdmPrimitiveTypeKind.Binary"

Reply:
Actually, I meant to call out that, if the stream was not known to be a string, then None would be returned, meaning that it *might* be binary.  I've clarified the comment to call out all three cases. 

---
In reply to: [258749210](https://github.com/OData/odata.net/pull/1204#discussion_r258749210) [](ancestors = 258749210)